pkg/ivd/ivd_protected_entity_type_manager_util.go

```diff
@@ -123,25 +135,30 @@ func findSharedDatastoresFromAllNodeVMs(ctx context.Context, client *vim25.Clien
 		return nil, errors.New("No hosts can be found for node VMs")
 	}
 
-	dcNameMap := make(map[string]bool)
+	dcPathMap := make(map[string]bool)
 	for _, host := range hosts {
-		dcName, err := findDataCenterFromAncestors(ctx, client, host.Reference(), logger)
+		dcPath, err := findDatacenterPath(ctx, client, host.Reference(), logger)
 		if err != nil {
 			logger.Debugf("Failed to find a datacenter from ancestors of VM, %v", host.Reference())
 			continue
 		}
-		_, ok := dcNameMap[dcName]
+		_, ok := dcPathMap[dcPath]
 		if !ok {
-			dcNameMap[dcName] = true
+			dcPathMap[dcPath] = true
```

Comment:
Do we need the check? The effect is same if you just did `dcPathMap[dcPath] = true` without checking if the key exists.

Reply:
Yes, you are right. Resolving it.